src/Datadog.Trace.ClrProfiler.Managed/Helpers/ReflectionHttpHeadersCollection.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ClrProfiler.Helpers
+{
+    internal readonly struct ReflectionHttpHeadersCollection : IHeadersCollection
```

Comment:
Now that it's isolated, it becomes clear that it would be possible implement a factory for those, that would cache the necessary Func<> objects so we don't have to fetch them every time. This would require some benchmarking to make sure it's worthy. Gonna add an item to the backlog for that.